tests/bdd/features/tokens/generate.feature

```diff
@@ -0,0 +1,69 @@
+Feature: Generate HTTP token for Repository Service for TUF (RSTUF)
+    As an admin,
+    Admin has deployed and bootstrapped RSTUF
+
+
+    Scenario Outline: Admin uses HTTP API to generate a token
+        Given the admin has the admin password
+        And the admin gets an 'access_token' by logging in to '/api/v1/token' with a 'write:token' scope
+        And the admin adds Authorization Bearer 'access_token' in the 'headers'
+        And the admin adds JSON payload with scopes: <scopes> and expires: <expires>
+        When the admin sends a POST request to '/api/v1/token/new'
+        Then the admin should get status code '200'
+        And the admin should get 'access_token' with a new token
+
+        Examples:
+            | scopes                                           | expires |
+            | ['write:targets']                                | 1       |
+            | ['write:targets', 'read:settings']               | 240     |
+            | ['read:bootstrap']                               | 1       |
+            | ['read:bootstrap','write:targets']               | 24      |
+            | ['read:settings']                                | 3       |
+            | ['read:token']                                   | 5       |
+            | ['write:targets', 'read:settings', 'read:tasks'] | 380     |
+
+
+   Scenario Outline: Admin cannot generate Token using HTTP API with an invalid expires
+        Given the admin has the admin password
+        And the admin gets an 'access_token' by logging in to '/api/v1/token' with a 'write:token' scope
+        And the admin adds Authorization Bearer 'access_token' in the 'headers'
+        And the admin adds JSON payload with scopes: <scopes> and expires: <expires>
+        When the admin sends a POST request to '/api/v1/token/new' with invalid 'expires' in 'payload'
+        Then the admin should get status code '422'
+
+        Examples:
+            | scopes                             | expires |
+            | ['read:bootstrap','write:targets'] | 0       |
+            | ['read:bootstrap','write:targets'] | -5      |
+            | ['read:bootstrap']                 | None    |
+
+
+   Scenario Outline: Admin cannot generate Token using HTTP API for certain scopes
+        Given the admin has the admin password
+        And the admin gets an 'access_token' by logging in to '/api/v1/token' with a 'write:token' scope
+        And the admin adds Authorization Bearer 'access_token' in the 'headers'
+        And the admin adds JSON payload with scopes: <scopes> and expires: <expires>
+        When the admin sends a POST request to '/api/v1/token/new' with not allowed 'scopes' in 'payload'
+        Then the admin should get status code '422'
```

Comment:
This is interesting, the tests do pass with the error code of `422` because the scopes given in the `Examples` fail the validation from the [`TokenRequestPayload`](https://github.com/repository-service-tuf/repository-service-tuf-api/blob/5204a7095f32672a927b41e4e5e975abd42eff38/repository_service_tuf_api/api/token.py#L58) Pydantic model.  In this case, we allow the admin to create new tokens (for distribution to others), but that can only have specific [limited scope](https://github.com/repository-service-tuf/repository-service-tuf-api/blob/3b40a7e44696f471a72ce7ef5b2dde2f0f2ea58c/repository_service_tuf_api/token.py#L88), and not the ability for example to `write:token` (which only admin should have). Thus, I think it makes sense that it's a validation error, since even with admin permissions you cannot create a token with these scopes.
In contrast, the `/api/v1/token` uses a `application/x-www-form-urlencoded` content type, and validation actually happens in [`token.post`](https://github.com/repository-service-tuf/repository-service-tuf-api/blob/3b40a7e44696f471a72ce7ef5b2dde2f0f2ea58c/repository_service_tuf_api/token.py#L170). For this case, you should not be able to create a token with scope that is not in the user.scopes, so it makes sense to be forbidden I think.

Reply:
Ahaaa, I see. That explains it.
It's not always visible where `pydanitc` validates the things, now when I look at it it's clear.